### PR TITLE
query functionality on its own

### PR DESCRIPTION
**Pull request by EsmaShr**
This is the version of gather and hunt together, not incorporated in the obsidian wrapper yet.

### CODE REVIEW

File: src/query.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+async function query(query, options = {}) {
```
Comment (dsymiller) [inline src/query.js:1]: We should probably put this in the ObsidianWrapper in gather_hunt.jsx like we did with the mutate functionality

File: src/query.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      );
+    }, pollInterval);
+    return interval;
+  }
+
+  // when cacheRead set to true
+  if (cacheRead) {
+    let resObj;
+    // when the developer decides to only utilize whole query for cache
+    if (wholeQuery) resObj = cache.readWholeQuery(query);
```
Comment (dsymiller) [inline src/query.js:28]: have we written this readWholeQuery method yet? we should probably hold off merging this until we have all of the functionality in place

**Review by dsymiller — CHANGES_REQUESTED** [review]
I think we should write the WholeQuery functionality and integrate everything into ObsidianWrapper before we merge

File: src/query.js (hunk 2 above)
Comment (EsmaShr) [reply src/query.js:28]: Yes, Eric added them on the cache class file, not submitted yet.

Comment (dsymiller) [reply src/query.js:28]: I think if possible we should have both things as part of the same pull request

File: src/query.js (hunk 1 above)
Comment (EsmaShr) [reply src/query.js:1]: This is just the query on its own, waiting for reviews on this implementation before incorporating it in the ObsidianWrapper.

Comment (dsymiller) [reply src/query.js:1]: gotcha!  I think overall the logic looks good

**Review by dsymiller — APPROVED** [review]
Looks good!

**Review by dsymiller — APPROVED** [review]

**Pull request merged**